### PR TITLE
Add accept to offer

### PR DESCRIPTION
**Pull request by kivine99**

### CODE REVIEW

File: src/objects/Offer.ts [hunk 1]
```diff
@@ -55,6 +55,12 @@ export class Offer implements IOffer {
     return tradeItems;
   }
 
+  //TODO: Implement this method
+  // https://github.com/InternetEnemies/combatcritters-ts/issues/61
+  public async accept(): Promise<boolean> {
+    return Math.random() < .5;
+  }
```
Comment (VanJackal) [inline src/objects/Offer.ts:62]: why would this be boolean, it should probably throw something if the purchase fails

**Review by VanJackal — CHANGES_REQUESTED** [review]

File: src/objects/Offer.ts (hunk 1 above)
Comment (kivine99) [reply src/objects/Offer.ts:62]: You wouldn't want to throw something if the purchase failed because you didn't have enough cards or currency

Comment (VanJackal) [reply src/objects/Offer.ts:62]: I feel like the front end should probably check if the user can purchase it and gray out the buy button if they cant

Comment (kivine99) [reply src/objects/Offer.ts:62]: Yea you're right. I think that logic should maybe be in the wrapper then. Have a function in offer that returns whether or not the user has the items for the offer

Comment (VanJackal) [reply src/objects/Offer.ts:62]: we might want it to be added to the compare function, have it return an object that has the missing items and also a `canPurchase` bool

Comment (kivine99) [reply src/objects/Offer.ts:62]: So have it return a 
```typescript
interface {
   tradeITems[]: ITradeItem<ICard | IPack | ICurrency>; 
   canPurchase: boolean;
 }
```

Comment (VanJackal) [reply src/objects/Offer.ts:62]: is `tradeItems` meant to be the missing items, then yeah

Comment (kivine99) [reply src/objects/Offer.ts:62]: yea I might try to think of a better name for that lol

**Review by VanJackal — APPROVED** [review]

**Review by VanJackal — APPROVED** [review]

**Pull request closed without merging**